Public/Src/Tools/DropDaemon/VsoClient.cs

```diff
@@ -347,9 +347,31 @@ private async Task<AssociationsStatus> AssociateAsync(FileBlobDescriptor[] blobs
                 abortIfAlreadyExists: false,
                 cancellationToken: Token).ConfigureAwait(false);
 
+            var result = associateResult.Item2;
+
             Interlocked.Add(ref Stats.TotalAssociateTimeMs, ElapsedMillis(startTime));
-            Interlocked.Add(ref Stats.NumFilesAssociated, blobsForAssociate.Length - associateResult.Item2.Missing.Count());
-            return associateResult.Item2;
+
+            var missingBlobIdsCount = associateResult.Item1.Count();
+            var associationsStatusMissingBlobsCount = associateResult.Item2.Missing.Count();
+            Interlocked.Add(ref Stats.NumFilesAssociated, blobsForAssociate.Length - missingBlobIdsCount);
+
+            if (missingBlobIdsCount != associationsStatusMissingBlobsCount)
+            {
+                m_logger.Warning("Mismatch in the number of missing files during Associate call -- missingBlobIdsCount={0}, associationsStatusMissingBlobsCount={1}",

```

Comment:
>Warning [](start = 25, length = 7)

Should this be a warning? Based on your theory this is just the way it works. Definitely log it for sake of debugging and to confirm our suspicion, but it probably deserves to be lower level like verbose